### PR TITLE
ZENKO-3643: added a refresh button for the data browser page

### PR DESCRIPTION
**Pull request by JulienMeziere**
I created a ListLayoutButtons and a ButtonsContainer to prepare for the future buttons that will have to be implemented according to the ticket's screenshot:

image

JIRA ticket: https://scality.atlassian.net/browse/ZENKO-3643

### CODE REVIEW

File: src/react/ui-elements/ListLayoutButtons.jsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    align-items: center;
+`;
+
+export function RefreshButton() {
+    const { bucketName } = useParams();
+    const { pathname } = useLocation();
+    const dispatch = useDispatch();
+    const isBrowsingObjects = !!matchPath(pathname, '/buckets/:bucketName/objects');
+    const prefixWithSlash = useMemo(() => {
+        const splitedPath = pathname.split('objects/');
```
Comment (naren-rajendran) [inline src/react/ui-elements/ListLayoutButtons.jsx]: small change, `splitedPath` -> `splittedPath`

Comment (JulienMeziere) [reply src/react/ui-elements/ListLayoutButtons.jsx]: Should be good now 👍️

**Review by Cuervino — APPROVED** [review]
Approved (following a live demo)

File: src/react/ui-elements/ListLayoutButtons.jsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
+import React, { useMemo } from 'react';
+import { matchPath, useLocation, useParams } from 'react-router-dom';
+import { Button } from '@scality/core-ui/dist/next';
+import { listBuckets } from '../actions/s3bucket';
+import { listObjects } from '../actions/s3object';
+import styled from 'styled-components';
+import { useDispatch } from 'react-redux';
+
+export const ButtonsContainer = styled.div`
```
Comment (nicolas2bert) [inline src/react/ui-elements/ListLayoutButtons.jsx]: Can this be moved to ui-elements/Container.jsx ?

File: src/react/ui-elements/ListLayoutButtons.jsx [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    const { bucketName } = useParams();
+    const { pathname } = useLocation();
+    const dispatch = useDispatch();
+    const isBrowsingObjects = !!matchPath(pathname, '/buckets/:bucketName/objects');
+    const prefixWithSlash = useMemo(() => {
+        const splittedPath = pathname.split('objects/');
+        if (splittedPath.length < 2 || splittedPath[1].length === 0) {
+            return '';
+        } else {
+            const prefix = splittedPath[1].slice(-1) === '/' ? splittedPath[1] : `${splittedPath[1]}/`;
```
Comment (nicolas2bert) [inline src/react/ui-elements/ListLayoutButtons.jsx]: You can use the `addTrailingSlash` utility method.

File: src/react/ui-elements/ListLayoutButtons.jsx [hunk 4]
```diff
@@ -0,0 +1,45 @@
+import React, { useMemo } from 'react';
```
Comment (nicolas2bert) [inline src/react/ui-elements/ListLayoutButtons.jsx]: This `Buttons` Redux container is not really an ui component and is very specific to data browser, so we should move it to `databrowser` directory.

Comment (JulienMeziere) [reply src/react/ui-elements/ListLayoutButtons.jsx]: I moved it in HeaderButtons inside the databrowser folder, is it good like this?

File: src/react/ui-elements/ListLayoutButtons.jsx (hunk 3 above)
Comment (JulienMeziere) [reply src/react/ui-elements/ListLayoutButtons.jsx]: Should be good now 👍️

File: src/react/ui-elements/ListLayoutButtons.jsx (hunk 2 above)
Comment (JulienMeziere) [reply src/react/ui-elements/ListLayoutButtons.jsx]: Should be good now 👍️

**Review by nicolas2bert — APPROVED** [review]
@Cuervino do you mind doing a last check since couple changes have been made? Thank you :)

**Comment by Cuervino** [comment]
> @Cuervino do you mind doing a last check since couple changes have been made? Thank you :)

Sure, I'll so it!

**Review by Cuervino — APPROVED** [review]
lgtm

**Pull request merged**